lib/compress/zstd_compress.c

```diff
@@ -6159,14 +6209,19 @@ size_t ZSTD_compressSequences(ZSTD_CCtx* const cctx, void* dst, size_t dstCapaci
  * @return : amount of data remaining to flush */
 size_t ZSTD_flushStream(ZSTD_CStream* zcs, ZSTD_outBuffer* output)
 {
-    ZSTD_inBuffer input = { NULL, 0, 0 };
+    ZSTD_inBuffer const nullInput = { NULL, 0, 0 };
+    int const stableInput = (zcs->appliedParams.inBufferMode == ZSTD_bm_stable);
+    ZSTD_inBuffer input = stableInput ? zcs->expectedInBuffer : nullInput;
+    input.size = input.pos; /* do not ingest more input during flush */
```

Comment:
Can this be an `assert(input.size == input.pos)` instead?

I believe that we should always have consumed the entire input, is that correct?

Reply:
Even with `ZSTD_c_stableInBuffer`, it's possible to meet a situation where some input has not been consumed.

This happens if output buffer size is not large enough, in which case, the streaming state machine will be stopped during a flush stage, with some input not yet consumed.
After that `ZSTD_flushStream()` does not present `inBuffer` anymore, so it's not possible to update `inBuffer->pos`. `ZSTD_flushStream()` can only flush what's left in the output buffer, not ingest a bit more. 
This situation is a bit different from `ZSTD_compressStream2(cctx, outbuf, inbuf, ZSTD_e_flush)`, which allows ingesting a bit more input as part of the flush operation.